### PR TITLE
use Patch API to add namespace annotations

### PR DESCRIPTION
**Pull request by chaitanyabandi**
While adding annotations to the namespace, using the Update API may result in
conflicts as "the object has been modified; please apply your changes to the
latest version and try again". Use Patch API to avoid this.

Signed-off-by: Chaitanya Bandi <kbandi@cs.stonybrook.edu>

#### What type of PR is this?
/kind bug
/kind flake

#### What this PR does / why we need it:

Fixes how the namespace is annotated in DaemonSet tests.
Avoids conflicts with other updates to the namespace.

#### Which issue(s) this PR fixes:

Similar to issue #100197 

#### Does this PR introduce a user-facing change?

```
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/area conformance

### CODE REVIEW

File: test/e2e/apps/daemon_set.go [hunk 1]
```diff
@@ -888,20 +889,20 @@ func clearDaemonSetNodeLabels(c clientset.Interface) error {
 func updateNamespaceAnnotations(c clientset.Interface, nsName string) (*v1.Namespace, error) {
```
Comment (dims) [inline test/e2e/apps/daemon_set.go:889]: if we go from update->patch, do we want to rename this method to reflect that?

Looks like it's called from only one spot:
https://cs.k8s.io/?q=updateNamespaceAnnotations&i=nope&files=&excludeFiles=&repos=kubernetes/kubernetes

Comment (chaitanyabandi) [reply test/e2e/apps/daemon_set.go]: Makes sense. Updated it, thank you.

**Comment by nikhita** [comment]
> While adding annotations to the namespace, using the Update API may result in
conflicts as "the object has been modified; please apply your changes to the
latest version and try again". Use Patch API to avoid this.

Trying to understand this better...if there are concurrent updates battling eachother, how does using patch _ensure_ that conflicts don't occur?

Would it be better to use `RetryOnConflict` instead?

See similar case for cronjobs:

https://github.com/kubernetes/kubernetes/blob/6e0e64c992439c66ec63a02e6d517928acc580d5/test/e2e/apps/cronjob.go#L364-L372

**Comment by chaitanyabandi** [comment]
> > While adding annotations to the namespace, using the Update API may result in
> > conflicts as "the object has been modified; please apply your changes to the
> > latest version and try again". Use Patch API to avoid this.
> 
> Trying to understand this better...if there are concurrent updates battling eachother, how does using patch _ensure_ that conflicts don't occur?
> 
> Would it be better to use `RetryOnConflict` instead?
> 

Thanks for taking a look.

Using the Patch API of the type Merge for setting annotations ensures that annotations provided in the call are merged with those that are already present in the resource. If the Update API is used, annotations that are already present would be replaced with those provided in the call. Replacing annotations may have undesirable side effects with other controllers that may be using them. Thus, using Patch seems like a better idea.

**Comment by chaitanyabandi** [comment]
@soltysh, Could you please look at this change when you get a chance? Thanks.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**